teps/0118-matrix-with-explicit-combinations-of-parameters.md

```diff
@@ -0,0 +1,415 @@
+The [golang-test][golang-test] `Task` defines one required parameter `"package"` to specify the base package to test, 
+and two optional `Parameters` (in addition to many others) `"GOARCH"` to specify the architecture target (the default 
+architecture is `amd64`) and `"flags"` to specify the flags for the `go test` command (the default `flags` are
+`-race -cover -v`).
+
+The `Pipeline` author can use `Matrix` with the `golang-test` `Task` to test a Go project on different `linux` platforms:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-to-build-and-test-go-project
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - ...
+    - name: golang-test
+      taskRef:
+        name: golang-test
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+      params:
+        - name: package
+          value: "github.com/tektoncd/pipeline"
+        - name: packages
+          value: "./pkg/..."
+      matrix:
+        - name: GOARCH
+          value:
+            - "linux/amd64"
+            - "linux/s390x"
+            - "linux/ppc64le"
+        - name: flags
+          value:
+            - "-race -cover -v" 
+    - ...
+```
+
+In this example, [golang-test][golang-test] will be executed on three different linux flavors with the single value
+of `flags` - `-race -cover -v`. However, `race` is only supported on `linux/amd64` and `linux/ppc64le` but not on 
+`linux/s390x`.
+
+Now, we need a way to only specify flags `-cover -v` for `linux/s390x`. The `Pipeline` will need to define a `Matrix` 
+that will execute `golang-test` in three different linux flavors. While running on `amd64` and `ppc64le`, `golang-test` 
+needs to execute `go test github.com/tektoncd/pipeline/pkg/... -race -cover -v`. While running on `linux/s390x`, 
+`golang-test` needs to executes `go test github.com/tektoncd/pipeline/pkg/... -cover -v`.
+
+#### Explicitly defining combinations in the Matrix
+
+Let's look at the same [kaniko][kaniko-task] `Task` from the  *Tekton Catalog*. The `Pipeline` author can use `Matrix`
+with the `kaniko` `Task` to build three separate images from a single repository:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-to-build-images-from-a-single-repo
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - ...
+    - name: kaniko-build
+      taskRef:
+        name: kaniko
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+      matrix:
+        - name: IMAGE
+          value: [ "image-1", "image-2", "image-3" ]
+    - ...
+```
+
+In this example, [Kaniko][kaniko-task] `Task` will be executed with three different values for `IMAGE` but with the
+same default value of the `DOCKERFILE` which is `./Dockerfile`.
+
+![Matrix for Kaniko with multiple Images](images/0118-matrix-kaniko-with-images.jpeg)
```

Comment:
NIT: jpeg is not the best choice for diagrams as it's not the best algorithm for fonts

Reply:
fixed, thanks!